src/DotNetty.Transport/Channels/Sockets/AbstractSocketChannel.Unsafe.cs

```diff
@@ -64,20 +68,21 @@ public sealed override Task ConnectAsync(EndPoint remoteAddress, EndPoint localA
 
                 try
                 {
-                    if (ch._connectPromise is object)
+                    if (ch._connectPromise is not null)
                     {
                         ThrowHelper.ThrowInvalidOperationException_ConnAttemptAlreadyMade();
                     }
 
                     bool wasActive = _channel.IsActive;
+                    var connectPromise = ch.NewPromise(remoteAddress);
```

Comment:
do we need to initialize _connectPromise field in case DoConnect returns true?

Reply:
changed